src/frontend/src/optimizer/plan_node/utils.rs

```diff
@@ -181,3 +181,26 @@ impl fmt::Debug for IndicesDisplay<'_> {
         f.finish()
     }
 }
+
+/// Call `debug_struct` on the given formatter to create a debug struct builder.
+/// If a property list is provided, properties in it will be added to the struct name according to
+/// the condition of that property.
+macro_rules! formatter_debug_plan_node {
+    ($formatter:ident, $name:literal) => {
+        $formatter.debug_struct($name)
+    };
+    ($formatter:ident, $name:literal, $( { $prop:literal, $cond:expr } ), *) => {
```

Comment:
```suggestion
    ($formatter:ident, $name:literal $(, { $prop:literal, $cond:expr } )* $(,)?) => {
```

Can we unify them in one branch?